### ISSUE TITLE
`argmax`/`argmin` hangs on GPU.

### ISSUE THREAD

**Issue by trivialfis** [nv-legate/cunumeric#1031]
**reproduce**
``` python
import cunumeric as cn

rng = cn.random.default_rng(1994)
n = 4096 * 32
y = rng.integers(low=0, high=4, size=int(n) * 2).reshape(int(n), 2)
y += y
idx = cn.argmax(y, axis=1)
print("idx", idx)
```

```
legate --gpus=1
```

**env**
cunumeric: 821d1c2d
legate.core: f1d4a3e

**Comment by trivialfis** [comment]
I have pinned it down to this line: https://github.com/nv-legate/cunumeric/blob/821d1c2d8efc2e1be6488796954241788b5d135b/src/cunumeric/unary/unary_red.cu#L294 .

**Comment by manopapad** [comment]
@aschaffer if you have some bandwidth, could you please take a look (or guide @trivialfis on what additional things to try)? First thing I would try is to remove the dependence on RNG, e.g. see if it reproduces with
```
n = 4096 * 32
y = cn.ones(int(n) * 2).reshape(int(n), 2)
y += y
idx = cn.argmax(y, axis=1)
```
or even
```
n = 4096 * 32
y = cn.ones((int(n), 2))
idx = cn.argmax(y, axis=1)
```

**Comment by trivialfis** [comment]
```
n = 4096 * 32
y = cn.ones((int(n), 2))
idx = cn.argmax(y, axis=1)
```
Yes, this hangs as well.

**Comment by trivialfis** [comment]
Turns out, my testing GPU is too old (<700), opened a PR for the fix: https://github.com/nv-legate/cunumeric/pull/1032 .

**Comment by m3vaz** [comment]
We do not support <700/SM70 (Volta). @manopapad can we close the issue and PR?

**Comment by trivialfis** [comment]
I'm fine with that. But it's probably best to let the compilation fail instead of using macro to silent the error. Also, the PR uses cub util for ptx code, which might be nice to have.

**Comment by magnatelee** [comment]
@trivialfis do you have a sense of how much slower `cub::MatchAny`  would be than `__match_any_sync`? if the gap isn't big, I guess it wouldn't be a bad idea to use `cub::MatchAny` for all archs.

**Comment by RAMitchell** [comment]
> We do not support <700/SM70 (Volta). @manopapad can we close the issue and PR?

I think its worth telling users that their GPU is not supported instead of hanging.

**Comment by magnatelee** [comment]
> I think its worth telling users that their GPU is not supported instead of hanging.

I agree, but we need to figure out the right place to add the check. @manopapad @bryevdv it feels like the right measure is to make `import cunumeric` fail if someone tries to do it with an unsupported architecture. what do you guys think?

**Comment by bryevdv** [comment]
@magnatelee That seems fine and simple to implement, but I also agree that having the compilation fail as well would be friendly for anyone building from source.

**Comment by magnatelee** [comment]
@bryevdv right, I think we need to update our build to catch this. @trxcllnt is it to require the gpu arch to be greater than or equal to volta in cmake?

**Comment by manopapad** [comment]
Currently Pascal is not officially supported, due to issues we have run into in the past with our kernels, some of which require independent thread scheduling. We have not done a comprehensive survey of our kernels to see which ones have this problem, so the stated policy in Build.MD is that we don't actively test on Pascal, but users are welcome to try it and report issues which we'll fix (it's on me to document this support level a bit more clearly). Do we want to change this policy to have pre-volta be completely unsupported?
